### PR TITLE
(improvement) Plan limit

### PR DESCRIPTION
**Pull request by Borfak**
Please describe your changes using the checklist below.

### Self-hosted support
- [ ] Your feature is implemented for the selfhosted version of Swetrix
- [x] This PR only updates the production code (e.g. paddle webhooks, CAPTCHA, blog, etc.)

### Database migrations
- [ ] Clickhouse / MySQL migrations added for this PR
- [ ] No table schemas changed in this PR

### Documentation
- [ ] You have updated the [documentation](https://github.com/swetrix/docs) according to your PR
- [x] This PR did not change any publicly documented endpoints

### CODE REVIEW

**Review by kruzhambus — APPROVED** [review]
Cool

**Review by Blaumaus — APPROVED** [review]

**Review by Blaumaus — CHANGES_REQUESTED** [review]
migrations

**Review by Blaumaus — CHANGES_REQUESTED** [review]
1. Add migrations for new user entity value:
```sql
ALTER TABLE user ADD column `planLimitNotificationPercentage` int NOT NULL DEFAULT '80';
```
2. In user controller (settings update) add an endpoint to update this value. Make sure to do validations that the value provided is:
 - `>=0`
 - `<=100`
 - `% 10 === 0` (i.e. 10, 20, 30, ... only allowed)

File: apps/production/src/user/dto/update-user.dto.ts [hunk 1]
```diff
@@ -21,4 +21,7 @@ export class UpdateUserProfileDTO {
 
   @ApiProperty({ example: '24-hour', enum: TimeFormat })
   timeFormat: string
+
+  @ApiProperty({ example: '80' })
+  planLimitNotificationPercentage: number
```
Comment (yevheniikulisidi) [inline apps/production/src/user/dto/update-user.dto.ts:26]: Name the field shorter, and add minimal validation.

File: apps/production/src/user/entities/user.entity.ts [hunk 2]
```diff
@@ -239,6 +240,9 @@ export class User {
   @Column({ default: false })
   trialReminderSent: boolean
 
+  @Column('int', { default: SEND_WARNING_AT_PERC })
+  planLimitNotificationPercentage: number
```
Comment (yevheniikulisidi) [inline apps/production/src/user/entities/user.entity.ts:244]: Name a shorter column.

File: apps/production/src/task-manager/task-manager.service.ts [hunk 3]
```diff
@@ -205,13 +215,25 @@ export class TaskManagerService {
...
+          {
+            amount: planLimitNotificationPercentage,
+            url: 'https://swetrix.com/billing',
+          },
           'broadcast',
         )
+        if (isTelegramChatIdConfirmed && telegramChatId) {
+          await this.telegramService.addMessage(
+            telegramChatId,
+            `You've reached ${planLimitNotificationPercentage}% of your monthly event limit.\nThat's a lot of traffic, congratulations!\nThat means that in case you use 100% of the available events until the end of the month, all the subsequent traffic will not be saved.\nTo avoid this we recommend that you upgrade your tier on the [Billing](https://swetrix.com/billing) page.`,
```
Comment (yevheniikulisidi) [inline apps/production/src/task-manager/task-manager.service.ts:231]: Add text hyphenation. It's not readable.

**Review by yevheniikulisidi — CHANGES_REQUESTED** [review]

**Comment by Blaumaus** [comment]
Closing as this is not really needed at the moment after the "locked dashboards" update

**Pull request closed without merging**